TaskLayer/SearchTask/SearchTask.cs

```diff
@@ -189,94 +190,105 @@ protected override MyTaskResults RunSpecific(string OutputFolder, List<DbForTask
                 ionTypes.Add(ProductType.C);
             TerminusType terminusType = ProductTypeToTerminusType.IdentifyTerminusType(ionTypes);
 
-            List<CompactPeptide> peptideIndex = null;
-            float[] keys = null;
-            List<int>[] fragmentIndex = null;
-            if (SearchType == SearchType.Modern || SearchType == SearchType.NonSpecific)
-            {
-                #region Generate indices for modern search
+            ParallelOptions parallelOptions = new ParallelOptions();
+            if (MaxDegreeOfParallelism.HasValue)
+                parallelOptions.MaxDegreeOfParallelism = MaxDegreeOfParallelism.Value;
+            MyFileManager myFileManager = new MyFileManager(DisposeOfFileWhenDone);
 
-                Status("Getting fragment dictionary...", new List<string> { taskId });
-                var indexEngine = new IndexingEngine(proteinList, variableModifications, fixedModifications, Protease, InitiatorMethionineBehavior, MaxMissedCleavages, MinPeptideLength, MaxPeptideLength, MaxModificationIsoforms, ionTypes, new List<string> { taskId });
-                string pathToFolderWithIndices = GetExistingFolderWithIndices(indexEngine, dbFilenameList);
+            Psm[][] globalPsms = new Psm[MassDiffAcceptors.Count()][];
 
-                Dictionary<float, List<int>> fragmentIndexDict;
-                if (pathToFolderWithIndices == null)
-                {
-                    var output_folderForIndices = GenerateOutputFolderForIndices(dbFilenameList);
-                    Status("Writing params...", new List<string> { taskId });
-                    WriteIndexEngineParams(indexEngine, Path.Combine(output_folderForIndices, "indexEngine.params"), taskId);
-
-                    Status("Running Index Engine...", new List<string> { taskId });
-                    var indexResults = (IndexingResults)indexEngine.Run();
-                    peptideIndex = indexResults.PeptideIndex;
-                    fragmentIndexDict = indexResults.FragmentIndexDict;
-
-                    Status("Writing peptide index...", new List<string> { taskId });
-                    WritePeptideIndex(peptideIndex, Path.Combine(output_folderForIndices, "peptideIndex.ind"), taskId);
-                    Status("Writing fragment index...", new List<string> { taskId });
-                    WriteFragmentIndexNetSerializer(fragmentIndexDict, Path.Combine(output_folderForIndices, "fragmentIndex.ind"), taskId);
-                }
-                else
+            for (int databaseSearchNumber = 0; databaseSearchNumber < NumberOfDatabaseSearches; databaseSearchNumber++)
+            {
+                List<Protein> proteinListSubset = proteinList.GetRange(databaseSearchNumber * proteinList.Count() / NumberOfDatabaseSearches, (databaseSearchNumber + 1) * proteinList.Count() / NumberOfDatabaseSearches);
+
+                List<CompactPeptide> peptideIndex = null;
+                float[] keys = null;
```

Comment:
Move to if (SearchType == SearchType.Modern || SearchType == SearchType.NonSpecific) statement below